### ISSUE TITLE
Developer Guide: How-to install Ubuntu Desktop for developers

### ISSUE THREAD

**Issue by rkratky** [canonical/open-documentation-academy#135]
### Context

Ubuntu Desktop for developers is a new documentation set that is geared towards application developers who wish to use Ubuntu Desktop as their development platform. It is not meant for developers who participate in the development of the Ubuntu distribution itself (nothing is stopping them from using the guidance the docs provide, but their use case is not what informs the strategy for the content of this docs set).

The Dev Guide focuses on work with six toolchains:

- GCC
- Python
- Java
- .NET
- Rust
- Golang

The basic structure of the docs set consists of two main parts:

- Generic guidance about Ubuntu Desktop installation and set up specifics for developers
- Toolchain-specific guidance on how to install and start using the indiv. toolchains and related tooling

### Task

Develop docs to populate the generic-guidance part of the docs set, i.e., the non-toolchain-specific parts of the Developer Guide. There are three parts to this:

- set up and installation
- version control
- IDEs

This task is about the first part: **How-to install Ubuntu Desktop for developers**.

A straightforward how-to guide (see [Diataxis how-to](https://diataxis.fr/how-to-guides/) and/or talk to @rkratky) on installing Ubuntu Desktop to serve as a developer workstation. This **SHOULD NOT** duplicate any existing (up-to-date) documentation on general installation. Links (or points in prerequisites) should be provided to this generic installation content.

The gist of this article should be:

* Basic Linux/Ubuntu devel workstation considerations (do you need/want GUI, dual-boot or not, dev-specific HW requirements, etc.)
* Deviations from a default install path (default vs extended selection? if extended - what? third-party repos and stuff?)
* Post-install set up -- any recommendations?

This article SHOULD NOT go into details about specific software, nor should it delve into instructions about indiv. toolchains. If any of these need mentioning (e.g. during 'extended install' selection), a link should be provided to the respective how-to that covers that topic in detail.

### Reference

- Diataxis: [How-to guides](https://diataxis.fr/how-to-guides/)
- GitHub: [ubuntu-for-developers-docs](https://github.com/canonical/ubuntu-for-developers-docs)

Note: This docs set is not yet published because it's still largely incomplete. If you decide to work on this issue, I will provide you with access details to be able to see the non-public version.

**Comment by davidekete** [comment]
Hi @rkratky, I'd like to work on this.

**Comment by rkratky** [comment]
Fantastic. I'll assign the issue to you. I see you also commented on the other two related issues. Would you like to work on all three simultaneously?

**Comment by davidekete** [comment]
> Fantastic. I'll assign the issue to you. I see you also commented on the other two related issues. Would you like to work on all three simultaneously?

Yes, I would like to work on the simultaneously, since it's a doc set, I though that would be best.

**Comment by davidekete** [comment]
> > Fantastic. I'll assign the issue to you. I see you also commented on the other two related issues. Would you like to work on all three simultaneously?
> 
> Yes, I would like to work on them simultaneously, since it's a doc set, I though that would be best.

**Comment by davidekete** [comment]
> > > Fantastic. I'll assign the issue to you. I see you also commented on the other two related issues. Would you like to work on all three simultaneously?
> > 
> > 
> > Yes, I would like to work on them simultaneously, since it's a doc set, I thought that would be best.

**Comment by rkratky** [comment]
> Yes, I would like to work on the simultaneously, since it's a doc set, I though that would be best.

Cool. I'm assigning them all to you then. Let me know if there's anything I can help with.

**Comment by davidekete** [comment]
> > Yes, I would like to work on the simultaneously, since it's a doc set, I though that would be best.
> 
> Cool. I'm assigning them all to you then. Let me know if there's anything I can help with.

No problem. Thank you.

**Comment by davidekete** [comment]
Hi @rkratky, I need some help with this issue. are you free anytime soon?

**Comment by rkratky** [comment]
Hi, thanks for reaching out. I'm afraid I won't really have time until the new year. That said, please, ask any questions you have, and I'll try to respond even before then.

**Comment by davidekete** [comment]
> Hi, thanks for reaching out. I'm afraid I won't really have time until the new year. That said, please, ask any questions you have, and I'll try to respond even before then.

Alright, thank you. My issue is primarily with how images for these types of guides are sourced. Do you get a testing device and get the screenshots of the process or do you use screenshots from previously done installations.

**Comment by davidekete** [comment]
Hi @rkratky, I'm done with this issue. Where do I submit a PR?

**Comment by rkratky** [comment]
Apologies for dragging my feet with the reply.

> Hi [@rkratky](https://github.com/rkratky), I'm done with this issue.

Fantastic!

> Where do I submit a PR?

The repo for the Developer Guide is at https://github.com/canonical/ubuntu-for-developers-docs (also linked from the issue description). Please, submit your PR there.

**Comment by rkratky** [comment]
PR: https://github.com/canonical/ubuntu-for-developers-docs/pull/13

**Comment by rkratky** [comment]
@davidekete, thank you for this contribution. It's a great start, and I appreciate all the effort that went into it.

I'll need to ask for some modifications to satisfy the requirements of this particular issue. Specifically, we'll need to remove some of the generic installation stuff and add information to guide the user with regard to developer-related considerations.

As mentioned in the definition of this issue:

_This SHOULD NOT duplicate any existing (up-to-date) documentation on general installation. Links (or points in prerequisites) should be provided to this generic installation content._

_The gist of this article should be:_

* _Basic Linux/Ubuntu devel workstation considerations (do you need/want GUI, dual-boot or not, dev-specific HW requirements, etc.)_
* _Deviations from a default install path (default vs extended selection? if extended - what? third-party repos and stuff?)_
* _Post-install set up -- any recommendations?_

First of all, we want to remove the generic installation instructions that duplicate documentation already available from [
Install Ubuntu desktop](https://ubuntu.com/tutorials/install-ubuntu-desktop#1-overview). That documentation provides a detailed description of all installation steps, and there's no need to have that info in our guide as well. Instead, our guide should just link to the available official doc.

What we _do_ need is guidance regarding potential differences from the standard installation procedure. While your doc touches upon some of these things, we need authoritative instructions and only about things that deviate from the default installation steps. It's fine if our doc is quite short.

Some general notes:

- Avoid using 'here' or similar non-specific link texts. See, for example, https://www.lamar.edu/web-communication/resources/avoid-using-click-here.html
- Do not link to unofficial (not published by Ubuntu/Canonical) docs directly from the guide; it's OK to include such links in a special section, such as **Additional resources**.

Please, let me know if I can be of any help.

**Comment by davidekete** [comment]
> [@davidekete](https://github.com/davidekete), thank you for this contribution. It's a great start, and I appreciate all the effort that went into it.
> 
> I'll need to ask for some modifications to satisfy the requirements of this particular issue. Specifically, we'll need to remove some of the generic installation stuff and add information to guide the user with regard to developer-related considerations.
> 
> As mentioned in the definition of this issue:
> 
> _This SHOULD NOT duplicate any existing (up-to-date) documentation on general installation. Links (or points in prerequisites) should be provided to this generic installation content._
> 
> _The gist of this article should be:_
> 
> * _Basic Linux/Ubuntu devel workstation considerations (do you need/want GUI, dual-boot or not, dev-specific HW requirements, etc.)_
> * _Deviations from a default install path (default vs extended selection? if extended - what? third-party repos and stuff?)_
> * _Post-install set up -- any recommendations?_
> 
> First of all, we want to remove the generic installation instructions that duplicate documentation already available from [ Install Ubuntu desktop](https://ubuntu.com/tutorials/install-ubuntu-desktop#1-overview). That documentation provides a detailed description of all installation steps, and there's no need to have that info in our guide as well. Instead, our guide should just link to the available official doc.
> 
> What we _do_ need is guidance regarding potential differences from the standard installation procedure. While your doc touches upon some of these things, we need authoritative instructions and only about things that deviate from the default installation steps. It's fine if our doc is quite short.
> 
> Some general notes:
> 
> * Avoid using 'here' or similar non-specific link texts. See, for example, https://www.lamar.edu/web-communication/resources/avoid-using-click-here.html
> * Do not link to unofficial (not published by Ubuntu/Canonical) docs directly from the guide; it's OK to include such links in a special section, such as **Additional resources**.
> 
> Please, let me know if I can be of any help.

Hi @rkratky, thanks for the corrections and general notes. I have updated the PR to reflect the changes you requested.

**Comment by rkratky** [comment]
@davidekete, apologies for taking so long with this. I have now submitted a PR against your topic branch at https://github.com/davidekete/ubuntu-for-developers-docs/pull/1

This is instead of a regular review because I ended up having to modify the article substantially. That's not because your submission would be wrong but because I realized the issue was badly scoped and described.

Most importantly, I moved the article from how-to to explanation. This is because as I was reviewing your submission and thinking about the desired outcome, I recognized that this article makes little sense as a how-to -- we're not trying to show instructions for installing the system. Instead, we need to explain how an installation of the system may differ for developers.

To that end, I also modified the content of the article.

When you merge the PR (linked above) I submitted against your branch, the [original PR](https://github.com/canonical/ubuntu-for-developers-docs/pull/13) will get up[dated and will be ready for merging into the guide.

**Comment by davidekete** [comment]
Hi @rkratky, thanks for the PR, I just merged it. The original should be good to go now.

**Comment by rkratky** [comment]
Done. Thank you for the great work (and patience), @davidekete!
